python/ray/autoscaler/v2/tests/test_scheduler.py

```diff
@@ -1247,7 +1247,14 @@ def test_outdated_nodes(disable_launch_config_check):
 
 @pytest.mark.parametrize("idle_timeout_s", [1, 2, 10])
 @pytest.mark.parametrize("has_resource_constraints", [True, False])
-def test_idle_termination(idle_timeout_s, has_resource_constraints):
```

Comment:
Besides `has_resource_constraints`, add `has_resource_requests` and `has_gang_resource_requests` to cover the situations where a node is needed by upcomping resource requests.